bugs/gh-234.txt

```diff
@@ -0,0 +1,847 @@
+=== Begin SRU Template ===
+[Impact]
+Some cloud environments pass multipart user-data as text/x-shellscript.
+We need to make sure that it is accepted by cloud-init as user-data.
+
+[Test Case]
+
+```
+#!/bin/sh
+set -x
+# Manually deploy on a lxc container
+
+echo -e 'The userdata-mime file was generated by running the command:
+tools/make-mime.py --attach userdata:x-shellscript  > userdata-mime
+
+where userdata is defined as the following file:
+#cloud-config
+final_message: The system is finally up, after seconds $UPTIME seconds'
```

Comment:
This is a good example @lucasmoura I think it would be nice to add this little mime example to the lxd FAQ section about some use-cases https://cloudinit.readthedocs.io/en/latest/topics/faq.html#lxd

Reply:
I know we have a part of those docs here https://cloudinit.readthedocs.io/en/latest/topics/format.html?highlight=make-mime#helper-script-to-generate-mime-messages

maybe it is worth cross referencing those docs too in the faq

Reply:
I think it is a good idea to add an example to the FAQ, but I don't know if this one is the best one, since it seems I little bit confusing at first, because the script is not shellscript per se, but the instance is passing this information with this type. Maybe the example should be one where the mime type is consistent with the content of the message. But I don't have much knowledge about MIME, so I don't know if my assumptions are correct